src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -176,6 +176,15 @@ public class CoinstakeContext
             public Key Key { get; set; }
         }
 
+        /// <summary>
+        /// Indicates the current state of the staking: in progress, stopped or idle.
+        /// </summary>
+        public enum CurrentState
```

Comment:
Why not just int with a comment that 0 is idle and 1 is executing? 

For example in network peer class we don't use enum and just int. 

Or we can add a class for atomic operations with bool. Example can be taken from here: https://stackoverflow.com/questions/6661055/using-interlocked-compareexchange-operation-on-a-bool-value/18027246

Reply:
I am on the fence here. Enums are much better for readability of the code and also allow you to extend states in the future, while booleans are prone to double negative usage which makes it hard to read and buggy, i.e. `stakeNotInProgress` and `!stakeNotInProgress` ... But I do see your point with casting to `int` which is a bit verbose, it would be nice if Interlocked.CompareExchenge supported enums.

Reply:
I actually like enum here